### ISSUE TITLE
Add possibility to schedule jobs

### ISSUE THREAD

**Issue by manisi** [jhipster/jhipster-dotnetcore#466]
##### **Overview of the feature request**

for execute certain taskin the background that should be executed at predefined intervals of time .

##### **Related issues or PR**

-   [ ] Checking this box is mandatory (this is just to show you read everything)

**Comment by nicolas63** [comment]
Do you need jhipster for that ? You can use  https://www.hangfire.io/.

**Comment by manisi** [comment]
> https://www.hangfire.io/

i think embed in jhipster generator like Quartz.net is simpler than hangfire. Thanks

**Comment by nicolas63** [comment]
Yes maybe, but what is the added value of generating this part ? Add the nuget and expose the configuration ? If it's just for that I think developers can do it

**Comment by manisi** [comment]
some time the first generated jhipster app need to use execute jobs. for example : the not active user must be deleted before 2 days ago. and the other hand yes add nuget package and use it. Thanks

**Comment by nicolas63** [comment]
If you need pre-made it's ok for me
Do you want work on it ?

**Comment by manisi** [comment]
> If you need pre-made it's ok for me
> Do you want work on it ?

ok ,i hope i can do it .

**Comment by nicolas63** [comment]
If you need help, don't hesitate to ping me

**Comment by maznag** [comment]
We can use integrated [.Net Core `HostedService`](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-5.0&tabs=visual-studio) to implement background task rather than use of external package 

And use of [`Cronos` package](https://www.nuget.org/packages/Cronos) to parse Cron expressions and calculating next occurrences.

**Comment by nicolas63** [comment]
the main advantage of the hangfire is that you can also have an admin part with the cron states, monitoring etc... 
The persistence of the jobs is also a good feature

**Comment by ivanmonteiro** [comment]
It would be interesting if the job scheduler could work properly when we horizontally scale the application (running the app on 3 servers, as example). We don't want the same job running on all servers. That would waste resources and lead to problems.

It seems that hangfire.io supports this scenario: https://docs.hangfire.io/en/latest/background-processing/running-multiple-server-instances.html

What I have not liked about hangfire.io is that the supported storage options are MSSQL and Redis only (https://docs.hangfire.io/en/latest/#job-storage). But I'm not sure it is a problem. I have not used hangfire.io before.

**Comment by ivanmonteiro** [comment]
Quartz.Net also seems interesting and worth investigating. It supports job persistence (in memory and database by ado.net), fail retry and, clustering (multi-server).
See: https://www.quartz-scheduler.net/

**Comment by ivanmonteiro** [comment]
My opinion is that the job shceduling solution should:

- be simple (no need of dependency on redis or external server at first)
- allow persistence - add robustness tor the application and is useful for server reboots, failover and retry
- work on multi-server applications. This is optional, but it is important for for future.

Both hangfire.io and Quartz.Net seems to have its pros and cons. 
Quartz.Net seems simple at first with in-memory persistence. 
Hangfire.io has more features (dashboard, etc) but the basic setup seems more complex as it needs eiher Redis or database persistence.

@maznag @nicolas63 You both have interesting ideas, please share what you think.
